base/server/src/main/java/org/dogtagpki/server/rest/UserService.java

```diff
@@ -1039,6 +1039,18 @@ public Response addUserMembership(String userID, String groupID) {
 
         if (userID == null) throw new BadRequestException("User ID is null.");
         if (groupID == null) throw new BadRequestException("Group ID is null.");
+        User user = null;
+
+        try {
+            user = userGroupManager.getUser(userID);
+        } catch (Exception e) {
+            throw new PKIException(getUserMessage("CMS_USRGRP_SRVLT_USER_NOT_EXIST", headers));
```

Comment:
The `UGSubsystem.getUser()` actually will return a `null` if the user doesn't exist:
https://github.com/dogtagpki/pki/blob/master/base/server/src/main/java/com/netscape/cmscore/usrgrp/UGSubsystem.java#L167-L169

So if the `UGSubsystem.getUser()` throws an exception, it must be caused by other issues, and we probably should pass that to the caller. What do you think?

Reply:
On this problem I have followed the same approach of other methods, as an example:

https://github.com/dogtagpki/pki/blob/master/base/server/src/main/java/org/dogtagpki/server/rest/UserService.java#L588-L592

https://github.com/dogtagpki/pki/blob/master/base/server/src/main/java/org/dogtagpki/server/rest/UserService.java#L645-L649

I think the behaviour should be the same when an error is throw  finding the user. If it is the case I can change this and separately we can update the others.

Reply:
Yeah, I think it would be misleading to just say "user doesn't exist" for any exception generated by `getUser()`. At least we should include the original exception message and attach the original exception too so the caller can see what's going on.

Feel free to merge this PR as is or with additional changes. The other can be fixed here or separately too. Thanks!